### DISCUSSION TITLE
Help regarding a custom loss for yolov8

### DISCUSSION THREAD

**Discussion by Shaurya-Rathore** [ultralytics/ultralytics#17189] [Q&A]
I am trying to run a custom loss function. Im getting the head output shape as 1,10,8400(6 classes + x y w h). Im unsure what the shapes of the targets and the predictions should be and what steps I am missing between 1,10,8400 to the required prediciton shape. Also in what manner should I be extracting the data from the yaml as in with what split.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Shaurya-Rathore, thank you for reaching out to Ultralytics 🚀! 

For customizing a loss function with YOLOv8, you're diving into an advanced topic, and we appreciate your interest and initiative! For your specific questions about shape transformations and target preparations, I recommend checking out our detailed [Docs](https://docs.ultralytics.com) for more insights into model architecture and data handling. 

If this is a 🐛 Bug Report, please provide a [Minimum Reproducible Example (MRE)](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us understand and potentially resolve any issues.

If this is a custom training ❓ question, providing additional context such as the dataset, example images, and any relevant logs can be helpful. Ensure you're considering our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/), which might offer some guidance on resolving common pitfalls.

Meanwhile, it might be helpful to connect with our vibrant Ultralytics community! For real-time assistance, you can join our [Discord](https://ultralytics.com/discord) 🎧. Alternatively, for more in-depth conversations, visit our [Discourse](https://community.ultralytics.com) or check out threads on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Make sure your `ultralytics` package is up-to-date, along with all dependencies, in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

Explore running YOLOv8 in various verified environments, which come with preinstalled dependencies like [CUDA](https://developer.nvidia.com/cuda), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. Refer to the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. Check out the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. Follow the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Your query will be addressed by an Ultralytics engineer soon to provide more specific guidance. Thanks again for your patience and interest in leveraging Ultralytics tools! 😊

**Comment by glenn-jocher** [comment]
@Shaurya-Rathore for custom loss functions in YOLOv8, ensure your predictions and targets match in shape. Predictions should be reshaped to match your target format, typically `[batch, num_anchors, num_classes + 4]`. Extract data from the YAML using the `data` argument in your training script. For more guidance, refer to the YOLOv8 documentation.
